### PR TITLE
Issue 693：CANNOT get profile data when creating sanpshot

### LINKED ISSUE #693: CANNOT get profile data when creating sanpshot
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug

**What happened**:
When creating snapshot, I call `pb.CreateVolumeSnapshotOpts.GetProfile()` to parse options
 of snapshot in my storage driver. But `pb.CreateVolumeSnapshotOpts.GetProfile()` returns nothing (not nil, it is string but blank).
```
#Profile
{
	"name": "nimble_snap_create01",
	"storageType": "hpe_nimble",
	"customProperties": {
			"online": false
	}
}
```
```
# Request Body for creating snapshot
{
  "name": "test-snap02",
  "description": "this is test",
  "volumeId": "ec22d73e-eede-417f-9e20-f59990ceaa72",
  "profileId": "044111ef-5ea4-412f-ae45-52dee991f11d"
}
```

**What you expected to happen**:
Available to get profile data by using `pb.CreateVolumeSnapshotOpts.GetProfile()` like `pb.CreateVolumeOpts.GetProfile()` .
`pb.CreateVolumeOpts.GetProfile()` is working good.

**How to reproduce it (as minimally and precisely as possible)**:
Using `pb.CreateVolumeSnapshotOpts.GetProfile()`.

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by BaiHuoYu**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #693 

**Special notes for your reviewer**:

**Release note**:

```Solve Issue [#693](https://github.com/opensds/opensds/issues/693)
```

### CODE REVIEW

**Comment by wisererik** [comment]
not only create snapshot, all rpc method should include profile detailed information

**Comment by leonwanghui** [comment]
> not only create snapshot, all rpc method should include profile detailed information

I talked to @xing-yang, and we agreed to just add it in `CreateVolume/DeleteVolume/ExtendVolume` and `CreateVolumeSnapshot/DeleteVolumeSnapshot` methods now.

File: pkg/api/controllers/volume.go [hunk 1]
```diff
@@ -686,6 +720,15 @@ func (v *VolumeSnapshotPortal) DeleteVolumeSnapshot() {
 		return
 	}
 
+	prf, err := db.C.GetProfile(ctx, snapshot.ProfileId)
```
Comment (leonwanghui) [inline pkg/api/controllers/volume.go]: This method should move ahead of `DeleteVolumeSnapshotDBEntry()` method.

**Comment by leonwanghui** [comment]
Can u rebase the code, it's out of date.

**Pull request merged**